anonymizer/src/main/java/gov/va/api/health/bulkfhir/anonymizer/ResourceBasedSyntheticData.java

```diff
@@ -0,0 +1,72 @@
+package gov.va.api.health.bulkfhir.anonymizer;
+
+import gov.va.api.health.dstu2.api.datatypes.HumanName;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+/** Class path resource based implementation of synthetic data. */
+@Slf4j
+@Builder
+public class ResourceBasedSyntheticData implements SyntheticData {
+
+  private final Names names;
+
+  /**
+   * We synthesize Dates and DateTimes by using: month = 1st month , day = first of the month This
+   * technique is a one way transformation and can't be reversed.
+   */
+  @Override
+  public String synthesizeDate(String rawDate) {
+    if (StringUtils.isBlank(rawDate)) {
+      return null;
+    }
+    LocalDate date;
+    try {
+      date = LocalDate.parse(rawDate);
+      date = date.withMonth(1);
+      date = date.withDayOfMonth(1);
+    } catch (DateTimeParseException e) {
+      log.info("Unable to parse the date [{}], using a default value.", rawDate);
+      date = LocalDate.of(2000, 1, 1);
+    }
+    return date.toString();
+  }
+
+  @Override
+  public String synthesizeDateTime(String rawDateTime) {
+    if (StringUtils.isBlank(rawDateTime)) {
+      return null;
+    }
+    LocalDateTime date;
+    try {
+      date = LocalDateTime.parse(rawDateTime);
```

Comment:
The time would get through the anonymization , and spelling of default in the log